Added Shorewall(6)-lite.

wvdakker opened the pull request:
Maintainer: Willem van den Akker/ <wvdakker@wilsoft.nl>
Compile tested: (ZyXel P2812-F1, LEDE Reboot)
Run tested: (ZyXel P2812-F1, LEDE Reboot, Install, configuration from admin system)

Description:
The Shoreline Firewall, is high-level tool for configuring Netfilter.

	Shorewall allows for central administration of multiple IPv4 firewalls
	through use of Shorewall lite. The full Shorewall product is installed
	on a central administrative system where compiled Shorewall scripts are
	generated. These scripts are copied to the firewall systems where they
	run under the control of Shorewall-lite.

diizzyy commented on net/shorewall-core/Makefile:
```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (C) 2008-2017 OpenWrt.org
```
https://forum.lede-project.org/t/packaging-guidelines/65/2
Please read "Copyright statements in Makefiles"

wvdakker replied:
The packages were part of BB. So the Makefiles are copied from BB.
The Copyright for OpenWrt will be the same. I added mine for 2017.

Hope to hear from you.

diizzyy commented on net/shorewall-core/Makefile:
```diff
@@ -0,0 +1,58 @@
+
+PKG_SOURCE_URL:=http://www.shorewall.net/pub/shorewall/$(PKG_DIRECTORY)/shorewall-$(PKG_MAINVERSION)/
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
+PKG_MD5SUM:=1912bf34776837dcca85741aeddd262a
```
Despite it's name please use SHA256 checksum here

wvdakker replied:
I would have if they are available. Only sha1sums or md5 can be used.

lucize commented:
@wvdakker can you make a pr for shorewall-full version ?
I'm using it on some complex multi isp (x86) and I installed it manually, some perl dependencies: perlbase-autoloader, perlbase-autouse, perlbase-dynaloader, perlbase-digest, perlbase-findbin, perlbase-getopt, perlbase-hash
regards

wvdakker commented:
@lucize: I will make this weekend a shorewall (full) package. It will be big because of all the perl dependencies. Not all router can run this full package.

lucize commented:
thanks,
the space usage for all the shorewall files including conffiles and macro from usr is about 2MB
I don't think size matters on usecases of shorewall

regards

wvdakker commented:
It takes more time to create an OpenWrt specific full Shorewall implementation. All install and configfiles must be 'OpenWrt' aware. 
It will be a seperate PR so it won't disturb the *lite transistion to OpenWrt.

lucize commented:
I have several futro s550 that use CF cards, and I use some industrial 128MB cards so I can't install a full system, they cand do ~800MB/s routing, high speed vnp too

regards

wvdakker commented:
Little tiny ping ping...
Someone willing to pull the request in?

lucize commented:
is there a reason not using 5.1.4 ?
I've made a full package based on your model, but I'm using the 5.1.4 version, I would like to add it, but it depends on shorewall-core and yours is 5.0.x version
The full version needs:         $(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/shorewall $(1)/usr/sbin found in core archive

lucize commented on net/shorewall-lite/files/vardir:
```diff
@@ -0,0 +1,3 @@
+# move state dir out of ram
+VARDIR=/etc/shorewall-lite/state
```
why not have it in tmpfs like /tmp/state ?

wvdakker replied:
Isn't /tmp removed after reboot? So I am missing the config files then. Or am I missing something.

lucize replied:
ah, I think that the lite is not compiling it at every start

wvdakker commented:
I have rebased the repos and removed the non-related commits. 

Running 5.1 forces you to install the Perl package. Even if you install the *lite version.

wvdakker commented:
Had today contact with one of the guys from Shorewall. 5.1 lite can be build without Perl package.
I have build the package. Will test it later this weekend and upload it.

hauke commented:
Please squash the commits together into one single commit

wvdakker commented:
Made a single commit.

hauke commented:
Please add the Signed-off-by line to the commit message again.

wvdakker commented:
Done
